app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt

```diff
@@ -1,34 +1,40 @@
 package com.willowtree.vocable.presets
 
 import android.content.Context
-import com.willowtree.vocable.room.Category
+import com.willowtree.vocable.room.CategoryDto
 import com.willowtree.vocable.room.Phrase
 import com.willowtree.vocable.room.VocableDatabase
+import kotlinx.coroutines.flow.Flow
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.get
-import java.util.*
+import java.util.Calendar
+import java.util.Locale
 
-class PresetsRepository(val context: Context) : KoinComponent {
+class PresetsRepository(val context: Context) : KoinComponent, IPresetsRepository {
 
     private val database = VocableDatabase.getVocableDatabase(context)
 
-    suspend fun getAllCategories(): List<Category> {
+    suspend fun getAllCategories(): List<CategoryDto> {
         return database.categoryDao().getAllCategories()
     }
 
-    suspend fun getUserGeneratedCategories(): List<Category> {
+    override fun getAllCategoriesFlow(): Flow<List<CategoryDto>> {
+        return database.categoryDao().getAllCategoriesFlow()
+    }
+
+    suspend fun getUserGeneratedCategories(): List<CategoryDto> {
         return database.categoryDao().getAllCategories()
     }
 
-    suspend fun getPhrasesForCategory(categoryId: String): List<Phrase> {
+    override suspend fun getPhrasesForCategory(categoryId: String): List<Phrase> {
         return database.categoryDao().getCategoryWithPhrases(categoryId)?.phrases ?: listOf()
     }
 
     suspend fun addPhrase(phrase: Phrase) {
         database.phraseDao().insertPhrase(phrase)
     }
 
-    suspend fun addPhraseToRecents(phrase: Phrase) {
+    override suspend fun addPhraseToRecents(phrase: Phrase) {
```

Comment:
do you still need to add phrase to recents if you changed he way we get them?

Reply:
Good question. I haven't changed the way we get them yet, I've only started differentiating the Category model that we store in the database (`CategoryDto`), from the Category domain model (`Category`)

Reply:
Getting myself back up to speed here 😅 I haven't actually even introduced `Category` yet in this code, only started adding the seams to do so, in this case `CategoriesUseCase`, which will eventually transform the database model into the domain model